python/pyspark/rdd.py

```diff
@@ -188,6 +188,12 @@ def __init__(self, jrdd, ctx, jrdd_deserializer=AutoBatchedSerializer(PickleSeri
         self._id = jrdd.id()
         self.partitioner = None
 
+    def __enter__(self):
```

Comment:
hmmm, yes this does happen to work, because most operations boil down to something like `mapPartitions` which creates a new `PipelineRDD` which is not cached, or a new `RDD` which is again not cached.

I think it will work for `DataFrame` too for similar reason - most operations return a new `DataFrame` instance.